wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,202 @@
+  constraints. 
+- Consider improving the pod lifecycle of DaemonSets and static pods during a node maintenance.
+- Explore the cloud provider use cases and how they can hook into the node lifecycle. So that the
+  users can use the same APIs or configurations across the board.
+- Migrate users of the eviction based kubectl-like drain (kubectl, cluster autoscaler, karpenter,
+  ...) and other scenarios to use the new unified node draining approach.
+- Explore possible scenarios behind the reason why the node was terminated/drained/killed and how to
+  track and react to each of them. Consider past discussions/historical perspective
+  (e.g. "tombstones").
+- Explore feedback mechanism for ensuring schedulability (e.g. readiness) and capabilities of the 
+  node. These can apply during the provisioning of the node, but also during the rest of the
+  node lifecycle.
+
+### Out of scope
+
+- Implementing cloud provider specific logic, the goal is to have high-level API that the providers
+  can use, hook into, or extend.
+- Infrastructure provisioning, deprovisioning solution or physical infrastructure lifecycle
+  management solution.
+
+## Stakeholders
+
+- SIG Apps
+- SIG Autoscaling
+- SIG CLI
+- SIG Cloud Provider
+- SIG Cluster Lifecycle
+- SIG Network
+- SIG Node
+- SIG Scheduling
+- SIG Storage
+
+Stakeholders span from multiple SIGs to a broad set of end users,
+public and private cloud providers, Kubernetes distribution providers,
+and cloud provider end-users. Here are some user stories:
+
+- As a cluster admin I want to have a simple interface to initiate a node drain/maintenance without
+  any required manual interventions.
+- As a cluster admin, I want to be able to observe the node drain via the API and check on its
+  progress. I also want to be able to discover workloads that are blocking the node
+  drain.
+- As a cluster admin, I want to be able to perform arbitrary actions after the node drain is
+  complete, such as resetting GPU drivers, resetting NICs, performing software updates or shutting
+  down the machine.
+- As a cluster admin, I want to reduce the cost of doing maintenance on my hardware accelerators by
+  using control-plane APIs to help coordinate maintenance and drain a Node.
+- To support the new features, node maintenance, scheduler, descheduler, pod autoscaling, kubelet,
+  and other actors should use a new eviction API to gracefully remove pods. This would enable new
+  migration strategies that prefer to surge (upscale) pods first rather than downscale them. It
+  would also allow other users/components to monitor pods that are gracefully removed/terminated
+  and provide better behaviour in terms of de/scheduling, scaling and availability.
+- As an end user, I would like more alternatives to blue-green upgrades, especially with special
+  hardware accelerators; it's far too expensive. I would like to choose a strategy on how to
```

Comment:
```suggestion
  hardware accelerators. I would like to choose a strategy on how to
```
That part of the statement doesn't provide any value to the use-case and it's just stating a fact, let's drop this.